imap_processing/hit/l1a/hit_l1a.py

```diff
@@ -50,203 +47,207 @@ def hit_l1a(packet_file: str, data_version: str) -> list[xr.Dataset]:
+
+    attr_mgr : ImapCdfAttributes
+        Attribute manager used to get the data product field's attributes.
 
     Returns
     -------
-    grouped_data : dict
-        Grouped data by apid.
+    dataset : xarray.Dataset
+        An updated dataset ready for CDF conversion.
     """
-    logger.debug("Grouping packet values for each apid")
-    grouped_data: dict = utils.group_by_apid(unpacked_data)
+    logger.info("Creating HIT L1A science datasets")
 
-    # Create data classes for each packet
-    for apid in grouped_data:
-        if apid == HitAPID.HIT_HSKP:
-            logger.debug(f"Grouping housekeeping packets - APID: {apid}")
-            grouped_data[apid] = [
-                Housekeeping(packet, "0.0", "hskp_sample.ccsds")
-                for packet in grouped_data[apid]
-            ]
-        else:
-            raise RuntimeError(f"Encountered unexpected APID [{apid}]")
+    # Logical sources for the two products.
+    # logical_sources = ["imap_hit_l1a_sci-counts", "imap_hit_l1a_pulse-height-event"]
+
+    # TODO: Complete this function
+    #  - call decom_hit.py to decommutate the science data
+    #  - split the science data into count rates and event datasets
+    #  - update dimensions and add attributes to the dataset and data arrays
+    #  - return list of two datasets (count rates and events)?
 
-    logger.debug("Finished grouping packet data")
-    return grouped_data
+    # logger.info("HIT L1A event dataset created")
+    # logger.info("HIT L1A count rates dataset created")
 
+    return dataset
 
-def create_datasets(data: dict, attr_mgr: ImapCdfAttributes) -> dict:
+
+def process_housekeeping(
+    dataset: xr.Dataset, attr_mgr: ImapCdfAttributes
+) -> xr.Dataset:
     """
-    Create a dataset for each APID in the data.
+    Will process housekeeping dataset for CDF product.
+
+    Updates the housekeeping dataset to replace with a single
+    leak_i variable as a 2D array. Also updates the dataset
+    attributes and coordinates and data variable dimensions
+    according to specifications in a cdf yaml file.
 
     Parameters
     ----------
-    data : dict
-        A single dictionary containing data for all instances of an APID.
+    dataset : xarray.Dataset
+        Dataset containing HIT housekeeping data.
+
     attr_mgr : ImapCdfAttributes
         Attribute manager used to get the data product field's attributes.
 
     Returns
     -------
-    processed_data : dict
-        A dictionary containing xarray.Dataset for each APID. Each dataset in the
-        dictionary will be converted to a CDF.
+    dataset : xarray.Dataset
+        An updated dataset ready for CDF conversion.
     """
-    logger.info("Creating datasets for HIT L1A data")
-
-    skip_keys = [
-        "shcoarse",
-        "ground_sw_version",
-        "packet_file_name",
-        "ccsds_header",
-        "leak_i_raw",
+    logger.info("Creating HIT L1A housekeeping dataset")
+
+    logical_source = "imap_hit_l1a_hk"
```

Comment:
Should this be in the CDF metadata?

Reply:
The logical source variable is used to grab the HIT L1A instrument level attributes from imap_hit_global_cdf_attrs.yaml when updating the dataset attributes `dataset.attrs = attr_mgr.get_global_attributes(logical_source)` 

Attributes from imap_hit_global_cdf_attrs.yaml for L1A housekeeping:
`imap_hit_l1a_hk:
    <<: *instrument_base
    Data_level: 1A
    Data_type: L1A_HK>Level-1A Housekeeping
    Logical_source: imap_hit_l1a_hk
    Logical_source_description: IMAP Mission HIT Instrument Level-1A Housekeeping Data.`

Would it help to rename this variable or plug in the value directly into the line that updates the dataset attributes? `dataset.attrs = attr_mgr.get_global_attributes(imap_hit_l1a_hk)`